workflow/controller/operator.go

```diff
@@ -186,10 +186,12 @@ func (woc *wfOperationCtx) operate(ctx context.Context) {
 	defer func() {
 		if r := recover(); r != nil {
 			woc.log.WithFields(log.Fields{"stack": string(debug.Stack()), "r": r}).Errorf("Recovered from panic")
-			if rerr, ok := r.(error); ok {
-				woc.markWorkflowError(ctx, rerr)
-			} else {
-				woc.markWorkflowError(ctx, fmt.Errorf("%v", r))
+			if !woc.wf.Status.Fulfilled() {
```

Comment:
I think we should error the workflow,  but only if the original workflow was not fulfilled.